test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -1276,28 +1276,292 @@ public void GetLink_TwoOptionalParametersAfterDefault_LastValueFromAmbientValues
+                {
+                    // link to different action on same controller
+                    {
+                        new { controller = "Products", action = "Edit", id = 10 },
+                        new { controller = "Products", action = "List" },
+                        new { controller = "Products", action = "List" },
+                        new { controller = "Products", action = "List" }
+                    },
+
+                    // link to different action on same controller and same area
+                    {
+                        new { area = "Customer", controller = "Products", action = "Edit", id = 10 },
+                        new { area = "Customer", controller = "Products", action = "List" },
+                        new { area = "Customer", controller = "Products", action = "List" },
+                        new { area = "Customer", controller = "Products", action = "List" }
+                    },
+
+                    // link from one area to a different one
+                    {
+                        new { area = "Admin", controller = "Products", action = "Edit", id = 10 },
+                        new { area = "Consumer", controller = "Products", action = "Edit" },
+                        new { area = "Consumer", controller = "Products", action = "Edit" },
+                        new { area = "Consumer", controller = "Products", action = "Edit" }
+                    },
+
+                    // link from non-area to a area one
+                    {
+                        new { controller = "Products", action = "Edit", id = 10 },
+                        new { area = "Consumer", controller = "Products", action = "Edit" },
+                        new { area = "Consumer", controller = "Products", action = "Edit" },
+                        new { area = "Consumer", controller = "Products", action = "Edit" }
+                    },
+
+                    // link from area to a non-area based action
+                    {
+                        new { area = "Admin", controller = "Products", action = "Edit", id = 10 },
+                        new { area = "", controller = "Products", action = "Edit" },
+                        new { area = "", controller = "Products", action = "Edit" },
+                        new { area = "", controller = "Products", action = "Edit" }
+                    },
+
+                    // link from controller-action to a page
+                    {
+                        new { controller = "Products", action = "Edit", id = 10 },
+                        new { page = "Products/Edit" },
+                        new { page = "Products/Edit"},
+                        new { page = "Products/Edit" }
+                    },
+
+                    // link from a page to controller-action
+                    {
+                        new { page = "Products/Edit", id = 10 },
+                        new { controller = "Products", action = "Edit" },
```

Comment:
Note that in a case like this the required values for a controller would be `new { controller = "Products", action = "Edit", page = (string) null },`

Reply:
True, I agree and `area = (string) null`